server/core/src/rodsAgent.cpp

```diff
@@ -271,7 +274,15 @@ runIrodsAgentFactory( sockaddr_un agent_addr ) {
             } else {
                 rodsLog( LOG_ERROR, "Agent process [%d] terminated with unusual status [%d]", reaped_pid, child_status );
             }
+
             rmProcLog( reaped_pid );
+
+            // TODO This should be scheduled on a thread pool to avoid blocking.
+            // Or maybe the replica access table allows marking PIDs for removal and
+            // invoking normal operations cause the table to clean-up before carrying out
+            // the operation.
+            ix::log::agent_factory::trace("Scheduling removal of agent PID [{}] from the replica access table ...", reaped_pid);
+            ix::replica_access_table::instance().erase_pid(reaped_pid);
```

Comment:
this needs to be scheduled on a thread pool.

Reply:
done.  however, there is the situation of multiple clients connecting to the same server and writing to it. this will lead to lock contention. a different approach may be needed to handle that (e.g. marking PIDs for removal and letting the thread pool erase the PIDs in bulk).

Reply:
until this becomes a problem, we'll let this solution ride.